Recast/Source/RecastArea.cpp

```diff
@@ -75,8 +58,8 @@ bool rcErodeWalkableArea(rcContext* ctx, int radius, rcCompactHeightfield& chf)
 						{
 							const int nx = x + rcGetDirOffsetX(dir);
 							const int ny = y + rcGetDirOffsetY(dir);
-							const int nidx = (int)chf.cells[nx+ny*w].index + rcGetCon(s, dir);
-							if (chf.areas[nidx] != RC_NULL_AREA)
+							const int nidx = (int)chf.cells[nx + ny*w].index + rcGetCon(s, dir);
+							if (chf.areas[nidx] != fromAreaID)
```

Comment:
I don't think this part is entirely correct. This will set the distance of a span to 0 not only when it has at least one neighbor with `fromAreaID`, but also when at least one of the neighbors is missing (`rcGetCon` returns `RC_NOT_CONNECTED`). That makes sense in the original use-case ("expanding" the unwalkable area), but isn't what we want when computing a general distance field.